src/main/scala/dsptools/numbers/algebra_types/Signed.scala

```diff
@@ -29,23 +29,21 @@ trait Signed[A] extends Any {
     val s = signum(a)
     !s.eq && !s.lt
   }
-  def isSignNegative(a: A): Bool = {
-    val s = signum(a)
-    !s.eq && s.lt
-  }
+  def isSignNegative(a: A): Bool = signum(a).lt
```

Comment:
@grebe double check changes here! Negative doesn't need !s.eq. Also, isSignNonPositive was originally incorrect (should be lt or eq)